components/Events/Filters.js

```diff
@@ -0,0 +1,222 @@
+import { faChevronUp, faChevronDown } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { useState, React } from 'react';
+import events from '../../data/event';
+import styles from '../../styles/components/Events/Filters.module.scss';
+
+export default function filters(props) {
+  const [isFilterOpen, setIsFilterOpen] = useState(false);
+  const [searchValue, setSearchValue] = useState('');
+  const [chosenComms, setChosenComms] = useState({
+    'studio': false,
+    'icpc': false,
+    'design': false,
+    'cyber': false,
+    'teach-la': false,
+    'w': false,
+    'ai': false,
+    'hack': false,
+    'board': false,
+  });
+  const [chosenDays, setChosenDays] = useState({
+    Sunday: false,
+    Monday: false,
+    Tuesday: false,
+    Wednesday: false,
+    Thursday: false,
+    Friday: false,
+    Saturday: false,
+  });
+  const [chosenTimes, setChosenTimes] = useState({
+    fromTime: '',
+    toTime: '',
+  });
+  const [chosenLoc, setChosenLoc] = useState({
+    'Online': false,
+    'In-Person': false,
+  });
+
+  const dateToMinutes = (date) => {
+    const itemMins =
+      date.getHours() * 60 + date.getMinutes();
+    return itemMins;
+  };
+
+  const handleSearch = () => {
+    // Check search input
+    const lowerCaseSearch = searchValue.toLowerCase();
+    let allEvents = events.filter((item) =>
+      [item.title, item.committee, item.location, item.description].some(
+        (itemsToSearch) =>
+          String(itemsToSearch).toLowerCase().includes(lowerCaseSearch),
+      ),
+    );
+
+    // Check time filter
+    const isFilterTimeEmpty = Object.keys(chosenTimes).every(
+      (time) => chosenTimes[time] === '',
+    );
+    if (!isFilterTimeEmpty) {
+      const filterFromMins =
+        parseInt(chosenTimes.fromTime.split(':')[0]) * 60 +
+        parseInt(chosenTimes.fromTime.split(':')[1]);
+      const filterToMins =
+        parseInt(chosenTimes.toTime.split(':')[0]) * 60 +
+        parseInt(chosenTimes.toTime.split(':')[1]);
```

Comment:
I think there's a possibility that `from` is set, but not `to` and I don't think this covers that edge case?